### DISCUSSION TITLE
3D buildings not working on macOS

### DISCUSSION THREAD

**Discussion by demiantres** [mapsforge/vtm#877] [Q&A] [answered]
I noticed that 3D buildings are not shown on macOS (they are shown in iOS). Is there a known workaround for this?

**Comment by devemux86** [comment, marked as answer]
OpenGL seems to be deprecated(?) on Apple OS (desktop + mobile).

As there are not many users using VTM on these platforms,
there are always problems with insufficient support / testing.
(compared to other mobile and desktop platforms)

You can search for macOS in the repository, play with the `BuildingLayer.TRANSLUCENT` flag, ...

**Reply by demiantres** [reply]
Thanks for the hints. I will check them out.

**Reply by demiantres** [reply]
BuildingLayer.TRANSLUCENT  did the trick 👍.

**Reply by devemux86** [reply]
> BuildingLayer.TRANSLUCENT did the trick +1.

What exactly?

There is already a check for macOS in [BuildingLayer](https://github.com/mapsforge/vtm/blob/master/vtm/src/org/oscim/layers/tile/buildings/BuildingLayer.java#L114-L116).
Does it need improvement?
